challenge_opener/src/main/java/com/example/challengeopener/ChallengeOpenerMainActivity.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.challengeopener
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.widget.Toolbar
+import com.example.challengeopener.databinding.ActivityChallengeOpenerMainBinding
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class ChallengeOpenerMainActivity : AppCompatActivity() {
+    private val binding by lazy { ActivityChallengeOpenerMainBinding.inflate(layoutInflater) }
```

Comment:
@js-gonzalez I'd suggest you avoid use lazy for binding. Keep in mind that it's necessary to release your variables to avoid leak memories.

I found interesting articles using the benefits of the Kotlin Property Delegation to handle view bindings on your activities, fragments, etc, please check them out:

https://proandroiddev.com/viewbinding-with-kotlin-property-delegate-c907682e24c9
https://blog.mindorks.com/property-delegation-in-kotlin
https://proandroiddev.com/make-android-view-binding-great-with-kotlin-b71dd9c87719